lib/ssh_scan/database/mongo.rb

```diff
@@ -2,36 +2,47 @@
 
 Mongo::Logger.logger.level = ::Logger::FATAL
 
-module   SSHScan
+module SSHScan
   module DB
     class MongoDb
-      def initialize(opts = {})
-        name = if !opts || opts.empty?
-                 'ssh_scan'
-               else
-                 opts["database"]["name"]
-               end
-        socket = "#{opts["database"]["server"]}:#{opts["database"]["port"]}"
-        @db = Mongo::Client.new([socket], :database => name)
-        @scans = @db[:scans]
+      attr_reader :collection
+
+      def initialize(client)
+        @client = client
+        @collection = @client[:ssh_scan]
+      end
+
+      # Helps us create a SSHScan::DB::MongoDB object with a hash
+      def self.from_hash(opts)
+        name = opts["name"]
+        server = opts["server"]
+        port = opts["port"]
+        socket = server + ":" + port.to_s
+
+        client = Mongo::Client.new([socket], :database => name)
+        return SSHScan::DB::MongoDb.new(client)
       end
 
+      # @param [String] worker_id
+      # @param [String] uuid
+      # @param [Hash] result
       def add_scan(worker_id, uuid, result)
-        @scans.insert_one("uuid" => uuid, "scan" => result,
+        @collection.insert_one("uuid" => uuid, "scan" => result,
                           "worker_id" => worker_id, "scanned_on" => Time.now)
       end
 
       def delete_scan(uuid)
-        @scans.delete_one(:uuid => uuid)
+        @collection.delete_one(:uuid => uuid)
       end
 
       def delete_all
-        @scans.delete_many({})
+        @collection.delete_many({})
       end
 
+      # LEFT OFF HERE: the results of this method should be the exact same format as with SQLite
```

Comment:
This is an area where this still needs work, the responses from find_scan_result should be the exact same format between mongodb and sqlite, they are not currently the case.  When this is fixed, the unit-tests can leverage shared specs so we know they will behave the same way.